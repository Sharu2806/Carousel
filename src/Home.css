.App {
  text-align: center;
  margin: 30px;
}


.header {
  padding: 20px;
  margin-bottom: 30px;
  background-color: #E9EFE7;
  text-align: left;
  font-weight: 600;
  font-size: 40px;
  color: #11669A;
}

.footer {
  padding: 30px;
  background-color: #E9EFE7;
  text-align: center;
  font-weight: 600;
  font-size: 40px;
  color: #11669A;
}

.footer button {
  padding: 10px 20px;
  background-color: #11669A;
  border: none;
  margin-right: 10px;
  color: #FFFFFF;
  font-size: 20px;
  font-weight: 600;
  font-family: initial;
}

.footer button:nth-child(1) {
  border-bottom-left-radius: 10px;
  border-top-left-radius: 10px;
}
.footer button:nth-child(2) {
  border-bottom-right-radius: 10px;
  border-top-right-radius: 10px;
}


.container {
  display: flex;
  overflow-x: hidden;
}
.container-cell {
  margin: 20px;
}

.container-cell img {
    width: 100%;
    height: 250px;
}

.imgTitle {
  color: #11669A;
  font-size: 20px;
  font-weight: 600;
  text-transform: capitalize;
}

@media only screen and (min-width: 600px) {
  .prevArrow,
  .nextArrow {
    display: none;
  }
}


/* Extra small devices (phones, 600px and down) */
@media only screen and (max-width: 600px) {
  .container-cell img{
    width: 410px;
    height: 300px;
  }
  .container-cell {
    margin: 0;
  }
  .prevArrow {
    background: url("./arrow.svg") no-repeat;
    width: 50px;
    height: 60px;
    transform: rotate(180deg);
    border: none;
    position: absolute;
    top: 30%;
    left: 5px;
  }
  .nextArrow {
    background: url("./arrow.svg") no-repeat;
    width: 50px;
    height: 60px;
    border: none;
    position: absolute;
    top: 30%;
    left: 85%;
  }
  .footer {
    display: none;
  }
  .App {
    margin: 0;
  }
}
