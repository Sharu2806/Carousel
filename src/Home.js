import React, { Component }  from 'react';
import { Header } from './Components'
import ImageContainer from './ImageContainer';
import './Home.css';
import axios from 'axios';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      allImages: []
    };
  }

  componentDidMount() {
    const that = this;
    axios.get('https://pixabay.com/api/?key=9656065-a4094594c34f9ac14c7fc4c39&q=beautiful+landscape&image_type=photo')
    .then(function (response) {
      that.setState(() => ({allImages: response.data.hits}));
    }).catch(function (error) {
      console.log(error);
    });
  }


  render() {
    const { allImages } = this.state;
    return (
      <div className="App">
        <Header />
        {(allImages.length > 0) && <ImageContainer allImages={allImages} />}
      </div>
    );
  }
}


export default Home;
