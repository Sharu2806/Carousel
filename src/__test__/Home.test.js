import React from 'react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from '../Home';

configure({ adapter: new Adapter() });
jest.mock('axios');
describe('App Component', () => {
  
  let data;
  beforeEach(() => {
    data = {
      hits: [
        "Image1.jpg",
        "Image2.jpg"
      ]
    }
    axios.get.mockImplementation(() => Promise.resolve({ data }));
  });

  test('should Match Screenshot', () => {
    const Component = renderer.create(<Home />);
    expect(Component.toJSON()).toMatchSnapshot();
  });

  test('should contains one footer', () => {
    const wrapper = mount(<Home />);
    wrapper.setState({allImages: data.hits});
    expect(wrapper.find('footer').length).toBe(1);
  });

  test('should contains one Header', () => {
    const wrapper = mount(<Home />);
    wrapper.setState({allImages: data.hits});
    expect(wrapper.find('header').length).toBe(1);
  });

  test('should fetch all images from given Url', () => {
    shallow(<Home />);
    expect(axios.get).toBeCalledWith(`https://pixabay.com/api/?key=9656065-a4094594c34f9ac14c7fc4c39&q=beautiful+landscape&image_type=photo`);
  });

  test('should call componentDidMount when mount()', () => {
    const componentDidMountSpy = jest.spyOn(Home.prototype, 'componentDidMount')
    mount(<Home />)
    expect(componentDidMountSpy).toHaveBeenCalledTimes(1)
    componentDidMountSpy.mockRestore()
  });
});
