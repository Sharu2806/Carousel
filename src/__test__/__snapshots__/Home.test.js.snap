// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should Match Screenshot 1`] = `
<div
  className="App"
>
  <header
    aria-label="Carousel Test Header"
    className="header"
    tabIndex="0"
  >
    Carousel Test
  </header>
</div>
`;
